### ISSUE TITLE
Completions pager down arrow bug on odd-numbered lists

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#4425]
Where an odd number of completions consuming multiple lines are available, navigating the completions via the down arrow results in a loop at the end, where fish jumps between the last two columns.

For example, after running this script:

```fish
mkdir completions_test
cd completions_test

for i in (seq 3 11)
    set fname ""
    for j in (seq 0 $i)
        set fname {$fname}a
    end
    echo $fname
    touch $fname
end
```

and then typing in `touch ./a<TAB><TAB>` in the resulting directory, the following list of completions appears (presuming your terminal is sized so that the results appear over two lines):

```fish
mqudsi@ZBook ~/r/f/completions_test> cat ./aaaa
         ./aaaaaa   ./aaaaaaaa   ./aaaaaaaaaa   ./aaaaaaaaaaaa
./aaaaa  ./aaaaaaa  ./aaaaaaaaa  ./aaaaaaaaaaa
```

If you press the down arrow until the last item in the list is selected, pressing the down arrow should either wrap to the start of the list or do nothing (since the end of the list has been reached). However, since in this case the last item in the list is on a higher screen line than the second-to-last item in the list, pressing the down arrow actually goes to the second entry of the second to last column (i.e. the second to last entry in the list). Since pressing the down arrow again would go to the next entry (the last entry), this continues _ad infinitum_.

**Comment by andrew-oxenburgh** [comment]
Just for information, I've run the test as you specified, and I can see what you mean. The cursor flickers back and forth. 

I'm not sure if it's a bug, but it does look odd. It could be argued that the cursor is selecting each final element in each row, but that's not for me to say.

**Comment by faho** [comment]
Fixed via 13079911bc4bbdd100b2216f884d82f2d13869a1.
